client.go

```diff
@@ -49,8 +50,13 @@ func NewRESTClient(cl *client.Harbor, authInfo runtime.ClientAuthInfoWriter) *RE
 
 // NewRESTClientForHost constructs a new REST client containing a swagger
 // API client using the defined host string + basePath as well as basic auth info.
-func NewRESTClientForHost(host, basePath, username, password string) *RESTClient {
-	swaggerClient := client.New(runtimeclient.New(host, basePath, nil), strfmt.Default)
+func NewRESTClientForHost(u, username, password string) *RESTClient {
+	url, err := url.Parse(u)
+	if err != nil {
+		panic(err)
```

Comment:
Since this is a library, I think it's better to leave the error handling to the caller. We should just return the error here.